packages/headless/src/controllers/facets/range-facet/date-facet/date-range.ts

```diff
@@ -0,0 +1,59 @@
+import dayjs from 'dayjs';
+import utc from 'dayjs/plugin/utc';
+import customParseFormat from 'dayjs/plugin/customParseFormat';
+import {DateRangeRequest} from '../../../../features/facets/range-facets/date-facet-set/interfaces/request';
+
+dayjs.extend(utc);
+dayjs.extend(customParseFormat);
+
+type DateType = string | number | Date;
+
+type DateOptions = {
+  useLocalTime?: boolean;
+  dateFormat?: string;
+};
+
+type DateRangeOptions = Partial<Omit<DateRangeRequest, 'start' | 'end'>> &
+  DateOptions & {
+    start: DateType;
+    end: DateType;
+  };
+
+export function isSearchApiDate(date: string) {
+  return formatForSearchApi(dayjs(date)) === date;
+}
+
+export function buildDateRange(config: DateRangeOptions): DateRangeRequest {
+  const start = buildDate(config.start, config);
+  const end = buildDate(config.end, config);
+
+  if (start === 'Invalid Date' || end === 'Invalid Date') {
```

Comment:
Could 'Invalid Date' be extracted to a variable? This path doesn't seem to be tested

Reply:
Also, is there something in `dayjs` we can use to verify that it's an `invalid date` instead of a string check like this ?

Because this seems kinda flaky.

Reply:
Good call, `dayjs` has an `isValid` method. I moved the throw into `buildDate` function, removing a path and adding which date we could not parse to the message.